Src/Newtonsoft.Json/Linq/JContainer.cs

```diff
@@ -959,9 +959,17 @@ string ITypedList.GetListName(PropertyDescriptor[] listAccessors)
 
         PropertyDescriptorCollection ITypedList.GetItemProperties(PropertyDescriptor[] listAccessors)
         {
+#if HAVE_APPCONTEXT
+            if (!MiscellaneousUtils.ComponentModelIsSupported)
+            {
+                throw new NotSupportedException(MiscellaneousUtils.ComponentModelNotSupportedMessage);
+            }
+#endif
             ICustomTypeDescriptor? d = First as ICustomTypeDescriptor;
 
+#pragma warning disable IL2026
```

Comment:
I don't think these feature switches are going to work like you want them to. What you are doing is disabling the static analysis at build time, so users don't get warnings up front. But they will still get warnings at publish time. And their app will fail at runtime becuase no one defaults `ComponentModelIsSupported = false` in the build.